clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -5446,9 +5463,15 @@ void CodeGenModule::EmitGlobalVarDefinition(const VarDecl *D,
     // type.
     if (RD && RD->hasAttr<SYCLAddIRAttributesGlobalVariableAttr>())
       AddGlobalSYCLIRAttributes(GV, RD);
-    // If VarDecl has a type decorated with SYCL device_global attribute, emit
-    // IR attribute 'sycl-unique-id'.
-    if (RD && RD->hasAttr<SYCLDeviceGlobalAttr>())
+    // If VarDecl has a type decorated with SYCL device_global attribute 
+    // emit IR attribute 'sycl-unique-id'.
+    if (RD && (RD->hasAttr<SYCLDeviceGlobalAttr>()))
+      addSYCLUniqueID(GV, D, Context);
+
+    // If VarDecl type is SYCLTypeAttr::host_pipe, emit the IR attribute 
+    // 'sycl-unique-id'.
+    auto Ty = D->getType();
+    if (isSyclType(Ty, SYCLTypeAttr::host_pipe))
```

Comment:
Why not
```suggestion
    if (isSyclType(D->getType(), SYCLTypeAttr::host_pipe))
```
I don't see more uses of `Ty`.